src/Umbraco.Web/Editors/CodeFileController.cs

```diff
@@ -0,0 +1,155 @@
+﻿using AutoMapper;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using Umbraco.Core.Models;
+using Umbraco.Web.Models.ContentEditing;
+using Umbraco.Web.Mvc;
+using Umbraco.Web.WebApi;
+
+namespace Umbraco.Web.Editors
+{
+    [PluginController("UmbracoApi")]
+    public class CodeFileController : UmbracoAuthorizedJsonController
+    {
+
+        public HttpResponseMessage PostCreate(string type, CodeFileDisplay display)
```

Comment:
Normally we have a `Save` model and a `Display` model so that any custom validation can be applied to specific models. In some cases this allows for greater flexibility. In this case this may not be necessary since these models are quite simple, but something to keep in mind.